### ISSUE TITLE
look for Dockerfile.envy in user repo with name of new environment

### ISSUE THREAD

**Issue by progrium** [progrium/envy#15]
Idea is you can have your existing project repos include a `Dockerfile.envy` that's a Dockerfile made specifically for building a development environment for that project. 

When you create a new environment in Envy, it will look for a repo on your github by that name and see if it has a `Dockerfile.envy`. If it does, it clones it as the env directory and builds with `Dockerfile.envy`. This means an env directory can have a Dockerfile or envy Dockerfile, and Envy will prefer ones with envy suffix. However, it will not clone repos with just a plain Dockerfile as to discourage bad practice.

**Comment by fgrehm** [comment]
> However, it will not clone repos with just a plain Dockerfile as to discourage bad practice.

Does that mean you'd not be up for having a user specific `Dockerfile` fallback for projects that don't have  a `Dockerfile{.envy}` around? One of my base images is capable of bringing up development environments for many different projects without the need to write project specific `Dockerfile`s and I'd love to use that with Envy paired with the support for arbitrary github projects.

**Comment by progrium** [comment]
Yeah I think I'll take that back. It can look for a `Dockerfile` if `Dockerfile.envy` doesn't exist.

**Comment by fgrehm** [comment]
Should we try a `Dockerfile.dev` before building with `Dockerfile`? I remember seeing people recommend a `Dockerfile.dev` for development specific images, so the lookup order would be like:
1. `Dockerfile.envy`
2. `Dockerfile.dev`
3. `Dockerfile`

**Comment by progrium** [comment]
The problem with dev or regular Dockerfiles is they're not necessarily made
for dev environments. My Dockerfile.dev are made to run the app in
development, not provide all the tooling needed to do all related tasks....

On Mon, Jul 6, 2015 at 12:51 PM, Fabio Rehm notifications@github.com
wrote:

> Should we try a Dockerfile.dev before building with Dockerfile? I
> remember seeing people recommend a Dockerfile.dev for development
> specific images, so the lookup order would be like:
> 1. Dockerfile.envy
> 2. Dockerfile.dev
> 3. Dockerfile
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/progrium/envy/issues/15#issuecomment-118937870.

## 

Jeff Lindsay
http://progrium.com

**Comment by fgrehm** [comment]
Gotcha!
